web_app/calendar/app.py

```diff
@@ -1,13 +1,36 @@
 import copy
-from flask import Flask
+from flask import Flask, request, redirect
 
 from helpers import gen_hours, gen_weeks, date_template
 from html_template import *
 
-
 app = Flask(__name__)
 
 
+@app.route(task_preset_link)
+@app.route(task_preset_link + '/')
+def get_form():
+    args = dict(request.args)
+    form = task_preset_form.format(redirect=args.get('redirect')[0],
+                                   year=args.get('y')[0],
+                                   month=args.get('m')[0],
+                                   day=args.get('d')[0],
+                                   hour=args.get('h')[0])
+    return form
+
+
+@app.route(task_creator_link, methods=['post'])
+def set_form():
+    print dict(request.form)
+    args = dict(request.args)
+
+    redirect_to_url = args.get('redirect', [task_preset_link])[0]
+    # print request.form
+    # resp = create_task(user_id='1', task_name=request.form.get('title')[0], date=None)
+
+    return redirect(redirect_to_url)
+
+
 @app.route(years_route)
 @app.route(years_route + '/')
```

Comment:
Please, add for all the routes allowed methods, because it is not clear, what methods are acceptable for one or another route

Reply:
I don't think, adding of the method "get" is default conf  and unnecessary in _routes_

Reply:
Beware, phrase "I don't think" means that you don't think at all :)